packages/feature-detector/src/find-globals.ts

```diff
@@ -0,0 +1,52 @@
+import { dirname, resolve } from 'path'
+import { Project, SourceFile, ts } from 'ts-morph'
+import { buildProject } from './utils/project'
+
+/**
+ * Find the list of globals used by source files in the provided project.
+ * Analyzed source files can be filtered by provided a list of glob patterns (default to all TypeScript and JavaScript files, excluding type definitions)
+ */
+export function findGlobals(
+  sourcePath: string,
+  project: Project = buildProject()
+): string[] {
+  const globals = new Set<string>()
+  const sourceFile = project.getSourceFileOrThrow(sourcePath)
+  addFileGlobals(sourceFile, globals)
+  for (const imported of sourceFile.getImportDeclarations()) {
+    console.log(imported.getModuleSpecifierSourceFileOrThrow())
+    const importedFile = project.getSourceFile(
```

Comment:
Does it visit imports in imported files recursively? Say we have following 3 files:

```ts
// index.ts
import * from './a';
```
```ts
// a.ts
import * from './b';
```
```ts
// b.ts
console.log(__dirname)
```

Does the current implementation detect `__dirname`?

Reply:
Sorry, see my previous comment: my experiment with imports does not work. Ideally, it would detect __dirname.

Reply:
I think analyzing a single file first will cover the majority of cases we want to detect. Let's improve incrementally later 😄